### PR TITLE
Bump kube-openapi to 20200410 version and SMD to V3 

### LINKED ISSUE #923: update k8s.io/kube-openapi to v0.0.0-20200403204345-e1beb1bd0f35
**What happened**:
update k8s.io/kube-openapi to v0.0.0-20200403204345-e1beb1bd0f35 
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
https://github.com/kubernetes/kubernetes/pull/80123
https://github.com/kubernetes/kubernetes/pull/89833
**Environment**:
- Arktos version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by sonyafenge**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:
This PR is part of kube-openapi upgrade: #923
Mainly reference from community kubernetes PR:
https://github.com/kubernetes/kubernetes/pull/79636
https://github.com/kubernetes/kubernetes/pull/79843
https://github.com/kubernetes/kubernetes/pull/80123
https://github.com/kubernetes/kubernetes/pull/79551
https://github.com/kubernetes/kubernetes/pull/80730
https://github.com/kubernetes/kubernetes/pull/77354
https://github.com/kubernetes/kubernetes/pull/87442

**Which issue(s) this PR fixes**:

Fixes #923 

**Special notes for your reviewer**:
Perf-test run with 100 hollow nodes finished with success for both density and load:
1. Scale-up logs can be found under GCP project: workload-controller-manager on sonya-uswest2: /home/sonyali/logs/perf-test/gce-100/arktos/pr1010-0304-100-up
2. Scale-out logs can be found under GCP project: workload-controller-manager on sonya-uswest2: /home/sonyali/logs/perf-test/gce-100/arktos/pr1010-0304-100-1tp

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: vendor/github.com/jstemmer/go-junit-report/formatter/BUILD [hunk 1]
```diff
@@ -2,15 +2,11 @@ load("@io_bazel_rules_go//go:def.bzl", "go_library")
...
 go_library(
     name = "go_default_library",
-    srcs = [
-        "doc.go",
-        "unstructured.go",
-        "value.go",
-    ],
-    importmap = "k8s.io/kubernetes/vendor/sigs.k8s.io/structured-merge-diff/value",
-    importpath = "sigs.k8s.io/structured-merge-diff/value",
+    srcs = ["formatter.go"],
```
Comment (Sindica) [inline vendor/github.com/jstemmer/go-junit-report/formatter/BUILD:5]: This does not look like a right file move

File: go.mod [hunk 2]
```diff
@@ -89,6 +89,8 @@ require (
 	github.com/heketi/tests v0.0.0-20151005000721-f3775cbcefd6 // indirect
 	github.com/jonboulle/clockwork v0.1.0
 	github.com/json-iterator/go v1.1.8
+	github.com/jstemmer/go-junit-report v0.0.0-20190106144839-af01ea7f8024
```
Comment (Sindica) [inline go.mod:92]: From the list of related K8s PRs, Update SMD to V3 was the last one to do. But this commit is the first in this PR. Are we sure this is the right order?

Comment (sonyafenge) [reply go.mod:92]: There are 5+ PRs in community before upgrade to SMD V3, and also another 5+ PRs for kube-openapi upgrade. some of them were doing upgrade at the beginning,  and some were doing upgrade at the end. I have merged all these vendor upgrade to one commit to reduce the complication of the PR.

File: vendor/github.com/jstemmer/go-junit-report/formatter/BUILD (hunk 1 above)
Comment (sonyafenge) [reply vendor/github.com/jstemmer/go-junit-report/formatter/BUILD:5]: This is auto generated by hack/update-vendor.sh. the correct process is remove "vendor/sigs.k8s.io/structured-merge-diff/value/BUILD". and new added file "vendor/github.com/jstemmer/go-junit-report/formatter/BUILD". not sure how git display this as file moving.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go [hunk 3]
```diff
@@ -123,44 +130,44 @@ func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string, o
...
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList).
 			Operation("list"+b.kind+"ForAllTenants"))
 		root = fmt.Sprintf("/apis/%s/%s/tenants/{tenant}/%s", b.group, b.version, b.plural)
 	}
 
-	routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList))
-	routes = append(routes, b.buildRoute(root, "", "POST", "create", sample).Reads(sample))
-	routes = append(routes, b.buildRoute(root, "", "DELETE", "deletecollection", status))
+	routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList))
```
Comment (h-w-chen) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:149]: are there any reasons that calling buildRoute(...) with one more arg than previous?

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/update-codegen.sh [hunk 4]
```diff
@@ -25,7 +25,7 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-
 # --output-base    because this script should also be able to run inside the vendor dir of
 #                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
 #                  instead of the $GOPATH directly. For normal projects this can be dropped.
-"${CODEGEN_PKG}/generate-groups.sh" all \
+bash "${CODEGEN_PKG}/generate-groups.sh" all \
```
Comment (h-w-chen) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/update-codegen.sh:31]: just curious - is bash mandatory here?

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go [hunk 5]
```diff
@@ -268,5 +268,21 @@ func (in *JSONSchemaProps) DeepCopy() *JSONSchemaProps {
 		}
 	}
 
+	if in.XListMapKeys != nil {
```
Comment (h-w-chen) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go:273]: (did not find a good place to raise question about this commit in whole; put at the fist line) understood this commit was cherry-picked from upstream. Is this commit absolutely needed for openapi/SMD version bumping, or just putting it in for convenience?

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go (hunk 3 above)
Comment (sonyafenge) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:149]: We cherry-picked testcase "TestCRDRouteParameterBuilder". Multi tenancy support and also community PR: https://github.com/kubernetes/kubernetes/pull/80074 are necessary to get this test case pass.

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/update-codegen.sh (hunk 4 above)
Comment (sonyafenge) [reply staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/update-codegen.sh:31]: this is coming from this community commit: https://github.com/kubernetes/kubernetes/pull/79081/commits/391da79abd7a7af1cfab04de35daba7cd58c4c80

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go (hunk 5 above)
Comment (sonyafenge) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go:273]: This is part of kube-openapi upgrade. more detail can be found in community PR: https://github.com/kubernetes/kubernetes/pull/77354

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go (hunk 3 above)
Comment (h-w-chen) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go:149]: thanks for the explanation. got it.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go (hunk 5 above)
Comment (h-w-chen) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/deepcopy.go:273]: ok; go for it.

**Review by h-w-chen — APPROVED** [review]
/lgtm

**Pull request merged**